extensions/api/auth-basic/build.gradle.kts

```diff
@@ -25,6 +25,15 @@ dependencies {
     api(project(":extensions:api:auth-spi"))
     api(project(":spi:core-spi"))
     implementation("jakarta.ws.rs:jakarta.ws.rs-api:${rsApi}")
+    implementation("org.junit.jupiter:junit-jupiter:5.8.1")
+    testImplementation("org.assertj:assertj-core:3.21.0")
+    testImplementation("org.mockito:mockito-inline:3.6.28")
+    testImplementation(project(":extensions:api:data-management:asset"))
```

Comment:
I understand that it might seem convenient to just use the Asset API as test dummy, but that's a bad idea. If the Asset API gets refactored, this would suddenly break tests in this module. We're violating the separation of concerns. Apart from that, it introduces a dependency across package boundaries, even if just for testing.

Although it might be more effort now, creating a simple test rest controller for testing is a lot cleaner, and will save us pain down the road, e.g. when new auth methods are added.

Reply:
The idea was to test if the AuthenticationRequestFilter works correctly together with the BasicAuthenticationService and the API would response with an appropriately status code, this is the reason why it was necessary to call a EDC runtime.

Regarding the other points, I rewrote the integration test to get rid of the Asset dependency and now there is a dummy API controller registered on the data management context.

Reply:
Ok, the `AuthenticationRequestFilter` is already covered in a test, and anything beyond that would effectively be testing Jetty (which I think we can assume to be working).
As for the `BasicAuthenticationService`, there is also a test for that.

In summary, I think this integration test adds little value compared to the impact on overall test runtime it imposes.

Reply:
This integration test shown some things that probably are not standardized/documented enough.
E.g., to have a fully working `AuthenticationService` an `ExceptionMapper` has to be explicitly defined, and that's not documented anywhere. 

Until those things are not documented/refactored I see the value brought by this IT

Reply:
Then this PR should have added documentation about the `ExceptionMapper`, etc. It would have been more valuable, and less effort at that.
Apart from that, the entire issue circumvented our planning process.